zio-interop-cats/shared/src/main/scala/zio/interop/cats.scala

```diff
@@ -290,6 +290,12 @@ private class CatsMonadError[R, E] extends MonadError[ZIO[R, E, *], E] with Stac
   override final def raiseError[A](e: E): ZIO[R, E, A] = ZIO.fail(e)
 
   override final def attempt[A](fa: ZIO[R, E, A]): ZIO[R, E, Either[E, A]] = fa.either
+
+  override def tailRecM[A, B](a: A)(f: A => ZIO[R, E, Either[A, B]]): ZIO[R, E, B] =
+    flatMap(ZIO.effectSuspendTotal(f(a))) {
+      case Left(a)  => tailRecM(a)(f)
+      case Right(b) => pure(b)
+    }
```

Comment:
I just realised that this might be a bit wasteful. Because we call `tailRecM` recursively that means we suspend on every iteration instead of just once. Do you think you could defined a local `loop` function without the suspend and the only suspend once initially? 🤔

Reply:
Something like this:
```scala
override def tailRecM[A, B](a: A)(f: A => ZIO[R, E, Either[A, B]]): ZIO[R, E, B] = {
    def loop(a: A): ZIO[R, E, B] = f(a).flatMap {
      case Left(a)  => loop(a)
      case Right(b) => ZIO.succeedNow(b)
    }

    ZIO.effectSuspendTotal(loop(a))
  }
```

Reply:
Yep, that sounds good!